pkg/serviceaccount/legacy.go

```diff
@@ -96,6 +107,27 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 		return nil, errors.New("sub claim is invalid")
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
```

Comment:
move this inside the `if v.lookup {` block, where we already have the secret and service account objects (and put it after all the lookup validation is done)

Reply:
done